docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/git-commit.md

```diff
@@ -67,6 +68,20 @@ steps:
 # Push, etc...
 ```
 
+### Skipped Commit When No Changes
+If there are no differences between the working tree and the current `HEAD` of the branch, 
+the `git-commit` step is skipped. This behavior avoids unnecessary empty commits.
+
+```
+steps:
+- uses: git-commit
+  config:
+    path: ./out
+    message: "Update application image"
+```
+In this example, if `./out` has no changes from the current HEAD of the branch, no commit will be created.
+The step will still succeed and output the SHA of the last commit.
+
```

Comment:
This change isn't really worthy of a whole section dedicated to it. Your one line change on line 9 seems sufficient.